embeddings/pipeline/lightning_classification.py

```diff
@@ -81,9 +81,11 @@ def __init__(
             load_dataset_kwargs=load_dataset_kwargs,
             **self.datamodule_kwargs
         )
+        datamodule.setup()
```

Comment:
We can either call it manually or ask user to give the num_classes argument to the pipelines since it is needed to initialize the lightning module

Reply:
https://github.com/PyTorchLightning/pytorch-lightning/issues/9943

we can do the workaround as described here

Reply:
I would bump the Pytorch Lightning version at first, and then made addaptation / corrections

Reply:
1.6 also introduces new deprecations so It would be nice to check it now

Reply:
> So we should bump version in this PR and check whether it's still occurring.

It does not matter if it will occur. According to the message from warning it will be called many times which will be waste of time. 

I like the workaround. Instead of it, can we perform `setup()` on `init`?

Reply:
Ok, after consideration of other issues, let's stick to 1.5.2 as for now. Sorry for that pushiness of 1.6, as I considered it as good moment for a bump as we haven't yet finished experiments. But considering other issues that would occur, the better solution is to stick with 1.5.2

Reply:
> > So we should bump version in this PR and check whether it's still occurring.
> 
> It does not matter if it will occur. According to the message from warning it will be called many times which will be waste of time.
> 
> I like the workaround. Instead of it, can we perform `setup()` on `init`?

Absolutely, we load the dataset anyway in `__init__` to save the dataset info thus we can just perform it once and utilize loaded dataset info 

> Ok, after consideration of other issues, let's stick to 1.5.2 as for now. Sorry for that pushiness of 1.6, as I considered it as good moment for a bump as we haven't yet finished experiments. But considering other issues that would occur, the better solution is to stick with 1.5.2

Ok agreed